sdk/test/UnitTests/Custom/Marshalling/StreamingUtf8JsonReaderTests.cs

```diff
@@ -0,0 +1,111 @@
+﻿using AWSSDK_DotNet.CommonTest.Utils;
+using AWSSDK_DotNet.UnitTests;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Amazon.Runtime.Internal.Util;
+using Amazon.EC2.Model;
+namespace AWSSDK.UnitTests
+{
+    /// <summary>
+    /// Protocol tests already exists to test the marshalling and unmarshalling of request and responses in json, but they don't test very
+    /// large payloads, which would trigger the logic for getting more bytes from the stream. 
+    /// This class just tests the wrapper class StreamingUtf8JsonReader. 
+    /// </summary>
+    [TestClass]
+    public class StreamingUtf8JsonReaderTests
```

Comment:
We should have a test for the scenario where the `PassReaderByRef` is used for converting the JsonDocument and the the document doesn't fit in the buffer.